app/sidekiq/pull_feeds_job.rb

```diff
@@ -0,0 +1,29 @@
+class PullFeedsJob
+  include Sidekiq::Job
+
+  def perform(*args)
+    # TODO: Make limit value configurable.
+    delete_all_non_recommended!
+    feed_items = prepare_feed_items_for_upsert
+    Feed.upsert_all(feed_items, returning: false, record_timestamps: true, unique_by: %i[feed_item_id source_id])
+  rescue => e
+    Rails.logger.error("Error pulling news feeds: #{e}")
+  end
+
+  private
+
+  # Adds our news feed source information - real world should
+  # get this from the database.
+  def prepare_feed_items_for_upsert
+    TopNewsService.new(limit: 500).execute.each do |feed_item|
+      feed_item.merge!(Feeds::HackerNews::Sourceable::SOURCE_H)
+    end
+  end
+
+  # Make sure we save our recommended feeds; purge the rest in
+  # prep for bringing in the new top stories.
+  def delete_all_non_recommended!
+    recommended_feed_ids = Feed.recommended_feed_items.pluck(:id)
+    Feed.where.not(id: recommended_feed_ids).destroy_all
+  end
+end
```

Comment:
This is a Sidekiq job designed to be run nightly using Sidekiq-chron. This job performs several actions 1) purges the feeds database of all feeds that have not been recommended (starred) by any user. 2) Retrieves all the top stories from our news sources, normalizes the data, and performs an "upsert". An upsert inserts new news items if they do not exist, and updates any news items that already exist. Feeds previously recommended (starred) are preserved and even updated with corrected information if necessary by the nightly feed.